doc/man/Makefile.am

```diff
@@ -0,0 +1,108 @@
+# Man pages are only built if the --enable-man-pages option was
+# passed to the configure script.
+#
+# They should always be built before creating a distribution tarball.
+
+# function which adds the source directory prefix and adds a given suffix
+manaddsuffix = $(addsuffix $(1),$(addprefix $(srcdir)/,$(2)))
+
+# list only the names without the .*.txt extension here:
+MAN1_NAMES = \
+	lttng-gen-tp
+MAN3_NAMES = \
+	lttng-ust \
+	tracef \
+	tracelog \
+	lttng-ust-dl \
+	lttng-ust-cyg-profile
+
+# troff man pages
+MAN3_TROFF = do_tracepoint.3 tracepoint.3 tracepoint_enabled.3
+
+# AsciiDoc sources and outputs
+MAN1_TXT = $(call manaddsuffix,.1.txt,$(MAN1_NAMES))
+MAN3_TXT = $(call manaddsuffix,.3.txt,$(MAN3_NAMES))
+MAN_TXT = $(MAN1_TXT) $(MAN3_TXT) $(MAN8_TXT)
+MAN_XML = $(patsubst $(srcdir)/%.txt,%.xml,$(MAN_TXT))
+
+# common AsciiDoc source files
+COMMON_TXT = \
+	$(srcdir)/common-footer.txt \
+	$(srcdir)/common-footer-author.txt \
+	$(srcdir)/log-levels.txt \
+	$(srcdir)/tracef-tracelog-limitations.txt
+
+# config
+ASCIIDOC_CONF = $(srcdir)/asciidoc.conf
+XSL_FILES = \
+	manpage.xsl \
+	manpage-callouts.xsl \
+	manpage-bold-literal.xsl \
+	manpage-links.xsl \
+	manpage-headings.xsl
+XSL_SRC_FILES = $(addprefix $(srcdir)/xsl/,$(XSL_FILES))
+
+# common dependencies
+COMMON_DEPS = $(ASCIIDOC_CONF) $(COMMON_TXT)
+
+# man pages destinations
+MAN1 = $(addsuffix .1,$(MAN1_NAMES))
+MAN3 = $(addsuffix .3,$(MAN3_NAMES))
+MAN = $(MAN1) $(MAN3)
+
+if MAN_PAGES_OPT
+# at this point, we know the user asked to build the man pages
+if HAVE_ASCIIDOC_XMLTO
+# tools
+ADOC = $(ASCIIDOC) -f $(ASCIIDOC_CONF) -d manpage \
+	-a lttng_version="$(PACKAGE_VERSION)"
+ADOC_DOCBOOK = $(ADOC) -b docbook
+XTO = $(XMLTO) -m $(firstword $(XSL_SRC_FILES)) man
+
+# recipes
+%.1.xml: $(srcdir)/%.1.txt $(COMMON_DEPS)
+	$(ADOC_DOCBOOK) -o $@ $<
+
+%.1: %.1.xml $(XSL_SRC_FILES)
+	$(XTO) $<
+
+%.3.xml: $(srcdir)/%.3.txt $(COMMON_DEPS)
+	$(ADOC_DOCBOOK) -o $@ $<
+
+%.3: %.3.xml $(XSL_SRC_FILES)
+	$(XTO) $<
+# only clean the generated files if we have the tools to generate them again
+CLEANFILES = $(MAN_XML) $(MAN)
+else # HAVE_ASCIIDOC_XMLTO
+# create man page targets used to stop the build if we want to
+# build the man pages, but we don't have the necessary tools to do so
+ERR_MSG = "Error: Cannot build target because asciidoc or xmlto tool is missing."
+ERR_MSG += "Make sure both tools are installed and run the configure script again."
+
+%.1: $(srcdir)/%.1.txt $(COMMON_DEPS)
+	@echo $(ERR_MSG)
+	@false
+
+%.3: $(srcdir)/%.3.txt $(COMMON_DEPS)
+	@echo $(ERR_MSG)
+	@false
+endif # HAVE_ASCIIDOC_XMLTO
+endif # MAN_PAGES_OPT
+
+# start with empty distributed/installed man pages
```

Comment:
This is a general comment about your comments: I usually do sentences (starts with a capital letter, ends with dot) everywhere.